event_sourcery/event_store/interfaces.py

```diff
@@ -46,3 +46,17 @@ def ensure_stream(self, stream_id: StreamId, versioning: Versioning) -> None:
     @abc.abstractmethod
     def delete_stream(self, stream_id: StreamId) -> None:
         pass
+
+    @abc.abstractmethod
+    def subscribe(
```

Comment:
@Enforcer Let's discuss this API for StorageStrategy.

I was assuming that passing correct subscription options would be handled by the `EventStore.subscribe` method. The backend will have optional parameters. But still what bothers me is that this method is not obvious. This data class `Subscription` was not much better. It was raising an exception for bad parameters, but I would like to avoid situations like this.
I was thinking about Factory objects that will have API that will limit options, and different configurations would have different types, so there won't be confusion. But it also seems to me over-engineering. WDYT

Reply:
Or maybe it's better to introduce separate method for every case? 
`subscribe_to_all(from_position: int)`
`subscribe_to_category(from_position: int | None, category: str)`
`subscribe_to_events(from_position: int | None, *events: str)`

@Enforcer WTD? I'm referring to backend API

Reply:
Oh, I like version with separate methods even better